webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -2916,18 +2926,33 @@ var userOverride = {
                     var kmChatInput = document.createElement(
                         preLeadCollection.element || 'input'
                     );
-                    kmChatInput.setAttribute(
-                        'id',
-                        'km-' + preLeadCollection.field.toLowerCase()
-                    );
-                    kmChatInput.setAttribute(
-                        'type',
-                        preLeadCollection.type || 'text'
-                    );
-                    kmChatInput.setAttribute(
-                        'name',
-                        'km-' + preLeadCollection.field.toLowerCase()
-                    );
+                    if (
+                        (preLeadCollection.type == 'number' &&
+                            preLeadCollection.field.toLowerCase() ==
+                                'mobile') ||
+                        'phone'
+                    ) {
+                        kmChatInput.setAttribute('id', 'km-phone');
+                        kmChatInput.setAttribute(
+                            'type',
+                            preLeadCollection.type
+                        );
+                        kmChatInput.setAttribute('pattern', '/^-?d+.?d*$/');
+                        kmChatInput.setAttribute(
+                            'onKeyPress',
+                            'if(this.value.length==10) return false;'
```

Comment:
phone numbers can be more than 10, check for the standard phone number max size. 
Another thing is, do we need to use setAttribute to set the javascript code? can this be done by add standard javascript code and javascript event for onkeypress